test/e2e/resize_nodes.go

```diff
@@ -421,6 +421,14 @@ var _ = Describe("Nodes", func() {
 			if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes, 10*time.Minute); err != nil {
 				Failf("Couldn't restore the original cluster size: %v", err)
 			}
+			// Many e2e tests assume that the cluster is fully healthy before they start.  Wait until
+			// the cluster is restored to health
+			By("waiting for system pods to successfully restart")
+			pods, err := framework.Client.Pods("kube-system").List(labels.Everything(), fields.Everything())
```

Comment:
Is the test always resizing the cluster up? If that's the case I agree with what you wrote. But I thought that we also have a test that is first downsizing the cluster and then restoring the original size by adding additional instance.